docs/user-assigned-msi-mode.md

```diff
@@ -4,7 +4,7 @@
 
 This option allows azure KeyVault to use the user assigned managed identity on the k8s cluster VMSS directly.
 
-AKS does not support user assigned managed identities yet, only system assigned managed identities. Until this gap is covered, use a service principal or system assigned identities with AKS.
+AKS does not support BYO managed identities yet. Until this gap is covered, use a service principal or the auto generated Kubelet MI with AKS.
```

Comment:
@TomGeske The user can still create and assign user managed identities manually to the cluster, right? Wondering what is BYO managed identity here?

Reply:
Just reading the section again. What was the intent of saying? Why are we calling SP at all?

> AKS does not support user assigned managed identities yet, only system assigned managed identities. Until this gap is covered, use a service principal or system assigned identities with AKS.

You have two options for AKS, no?

1. Use Kubletidenity (user assigned but doesn't support BYO)
2. Create your own MI and assign it according to step 1 and 2

Reply:
@TomGeske that's my understanding too. Summarizing the options for available modes -

1. SP (use dedicated SP for access only to kv)
2. pod-identity (use MSI which will be assigned and managed by pod-identity)
3. User Managed Identity (Use AKS kubelet identity/ create own identity and assign manually, which is already documented. I think we need to remove the this entire snippet `AKS does not support user assigned managed identities yet, only system assigned managed identities. Until this gap is covered, use a service principal or system assigned identities with AKS.`
4. System assigned MSI

We have this in doc for system assigned MSI, but I don't think is correct -
```
For AKS clusters, system assigned managed identities can be created on your behalf. To learn more read here about AKS managed identity support.
```
There is only User MI created, right?

Reply:
Yes, similar to your documents. 

1. [Service Principal](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/service-principal-mode.md)
   * LGTM
2. [Pod Identity](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/pod-identity-mode.md)
   * LGTM
3. [VMSS User Assigned Managed Identity](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/user-assigned-msi-mode.md)
   * For AKS you can user your Kublet MI (doesn't support BYO today) or your assign manually your own MI.
4. [VMSS System Assigned Managed Identity](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/system-assigned-msi-mode.md)
   * Doesn't make sense for AKS

Makes sense?

Reply:
@TomGeske That makes sense! Would you be interested in updating this PR?